examples/cacoh.py

```diff
@@ -0,0 +1,496 @@
+"""
+========================================
+Compute multivariate coherency/coherence
+========================================
+
+This example demonstrates how canonical coherency (CaCoh)
+:footcite:`VidaurreEtAl2019` - a multivariate method based on coherency - can
+be used to compute connectivity between whole sets of sensors, alongside
+spatial patterns of the connectivity.
+"""
+
+# Authors: Mohammad Orabe <orabe.mhd@gmail.com>
+#          Thomas S. Binns <t.s.binns@outlook.com>
+# License: BSD (3-clause)
+
+# %%
+import numpy as np
+from matplotlib import pyplot as plt
+
+import mne
+from mne_connectivity import seed_target_indices, spectral_connectivity_epochs
+
+###############################################################################
+# Background
+# ----------
+#
+# Multivariate forms of signal analysis allow you to simultaneously consider
+# the activity of multiple signals. In the case of connectivity, the
+# interaction between multiple sensors can be analysed at once, producing a
+# single connectivity spectrum. This approach brings not only practical
+# benefits (e.g. easier interpretability of results from the dimensionality
+# reduction), but can also offer methodological improvements (e.g. enhanced
+# signal-to-noise ratio).
+#
+# A popular bivariate measure of connectivity is coherency/coherence, which
+# looks at the correlation between two signals in the frequency domain.
+# However, in cases where interactions between multiple signals are of
+# interest, computing connectivity between all possible combinations of signals
+# leads to a very large number of results which is difficult to interpret. A
+# common approach is to average results across these connections, however this
+# risks reducing the signal-to-noise ratio of results and burying interactions
+# that are present between only a small number of channels.

```

Comment:
Does multivariate not estimate all possible combinations of signal's connectivity? Perhaps we can clarify a bit more how bivariate differs from multivariate explicitly?

Reply:
I wouldn't say the multivariate methods estimate all possible combinations of connectivity, rather they involve a transformation from sensor space to an underlying component space according to the strongest interactions present.

Perhaps I can rephrase the following paragraph to something like:
```
# Canonical coherency (CaCoh) is a multivariate form of coherency that uses
# eigendecomposition-derived spatial filters to extract the underlying
# components of connectivity in a frequency-resolved manner
# :footcite:`VidaurreEtAl2019`. This approach goes beyond simply aggregating
# information across all possible combinations of signals.
#
# It is similar to multivariate methods based on the imaginary part of
# coherency (MIC & MIM :footcite:`EwaldEtAl2012`; see :doc:`mic_mim` and
# :doc:`compare_coherency_methods`).
```

Reply:
Sure.